plugins/debuginfo-install.py

```diff
@@ -79,54 +83,95 @@ def run(self):
         errors_spec = []
 
         for pkgspec in self.opts.package:
-            package_query = sorted(dnf.subject.Subject(pkgspec).get_best_query(
-                self.cli.base.sack).filter(arch__neq='src'), reverse=True)
-            if not package_query:
-                msg = _('No match for argument: %s')
-                logger.info(msg, self.base.output.term.bold(pkgspec))
+            solution = dnf.subject.Subject(pkgspec).get_best_solution(self.base.sack)
+
+            query = solution["query"].filterm(arch__neq='src')
+
+            packages = query.available()._na_dict()
+            # installed versions of packages have priority, replace / add them to the dict
+            packages.update(query.installed()._na_dict())
+
+            # Remove debuginfo packages if their base packages are in the query.
+            # They can get there through globs and they break the installation
+            # of debug packages with the same version as the installed base
+            # packages. If the base package of a debuginfo package is not in
+            # the query, the user specified a debug package on the command
+            # line. We don't want to ignore those, so we will install them.
+            # But, in this case the version will not be matched to the
+            # installed version of the base package, as that would require
+            # another query and is further complicated if the user specifies a
+            # version himself etc.
+            for na in list(packages.keys()):
+                if na[0].endswith(Package.DEBUGINFO_SUFFIX):
```

Comment:
Providing a constant by Package.DEBUGINFO_SUFFIX is different way that was used so far for name of system repo or so on.

Reply:
I'm not sure what you mean :confused:

Reply:
What about to put DEBUGINFO_SUFFIX outside of package class

Reply:
Why?

Reply:
Debuginfo surfix is identical for package and for repo.

Reply:
Anyway add local variable debuginfo_surfix = Package.DEBUGINFO_SUFFIX. I believe that it is a better to not call package object to get static value.

Reply:
Hmm, sorry, still not entirely clear.. so in stead of:

```python
class Package:
    DEBUGINFO_SUFFIX = "-debuginfo"
    # ...
```

You want me to put the constant straight into the module like:
```python
DEBUGINFO_SUFFIX = "-debuginfo"

class Package:
    # ...
```

At least that's what I think makes sense and I can do that... But it doesn't seem exactly what you're suggesting?

Reply:
Yes, that what I suggested. The question is in which module in libdnf.

Reply:
I clearly need it for `package` here. IIUC you're saying the suffix is the same for repositories, but that is sort of a coincidence and it's also just a convention? We shouldn't try to link the two I think. I'd put it in `package`. I kind of liked it in the class, since that's the one working with the suffixes and it kinda feels it belongs there...

Reply:
Please make change according to your best opinion

Reply:
After some discussion with @m-blaha I'll keep the constants inside the class.